processing/src/main/java/org/apache/druid/query/aggregation/ExpressionLambdaAggregator.java

```diff
@@ -89,4 +96,88 @@ public void close()
   {
     // nothing to close
   }
+
+  /**
+   * Tries to mimic the byte serialization of {@link Types} binary methods use to write expression values for the
+   * {@link ExpressionLambdaBufferAggregator} in an attempt to provide consistent size limits when using the heap
+   * based algorithm.
+   */
+  @VisibleForTesting
+  public static void estimateAndCheckMaxBytes(ExprEval eval, int maxSizeBytes)
```

Comment:
nit: I feel perhaps there is a better place for this method since it's nice to have size estimation methods and serialization methods in the same class so that we won't forget the other when we modify one of them. But maybe it's fine to be here since it's used only in this class.

Reply:
Maybe package-private is better.

Reply:
if I do the `TypeStrategy` thing mentioned in the other comment, then size estimation will be pushed there which seems like a more appropriate place. I put it here for now because this is the only caller